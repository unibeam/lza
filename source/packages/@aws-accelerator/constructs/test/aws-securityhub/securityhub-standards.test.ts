/**
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 *  with the License. A copy of the License is located at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
 *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
 *  and limitations under the License.
 */

import * as cdk from 'aws-cdk-lib';
import { SecurityHubStandards } from '../../index';
import { snapShotTest } from '../snapshot-test';

const testNamePrefix = 'Construct(SecurityHubStandards): ';

//Initialize stack for snapshot test and resource configuration test
const stack = new cdk.Stack();

new SecurityHubStandards(stack, 'SecurityHubStandards', {
  standards: [
    {
      name: 'AWS Foundational Security Best Practices v1.0.0',
      enable: true,
      controlsToDisable: ['IAM.1', 'EC2.10', 'Lambda.4'],
    },
    {
      name: 'PCI DSS v3.2.1',
      enable: true,
      controlsToDisable: ['IAM.1', 'EC2.10', 'Lambda.4'],
    },
  ],
  kmsKey: new cdk.aws_kms.Key(stack, 'CustomKey', {}),
  logRetentionInDays: 3653,
});

/**
 * SecurityHubStandards construct test
 */
describe('SecurityHubStandards', () => {
  snapShotTest(testNamePrefix, stack);
});
