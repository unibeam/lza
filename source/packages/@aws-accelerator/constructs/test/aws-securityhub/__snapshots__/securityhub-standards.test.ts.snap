// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecurityHubStandards Construct(SecurityHubStandards):  Snapshot Test 1`] = `
{
  "Resources": {
    "CustomKey1E6D0D07": {
      "DeletionPolicy": "Retain",
      "Properties": {
        "KeyPolicy": {
          "Statement": [
            {
              "Action": "kms:*",
              "Effect": "Allow",
              "Principal": {
                "AWS": {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition",
                      },
                      ":iam::",
                      {
                        "Ref": "AWS::AccountId",
                      },
                      ":root",
                    ],
                  ],
                },
              },
              "Resource": "*",
            },
          ],
          "Version": "2012-10-17",
        },
      },
      "Type": "AWS::KMS::Key",
      "UpdateReplacePolicy": "Retain",
    },
    "CustomSecurityHubBatchEnableStandardsCustomResourceProviderHandler4BE622C1": {
      "DependsOn": [
        "CustomSecurityHubBatchEnableStandardsCustomResourceProviderRole1ABC8ED2",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "REPLACED-GENERATED-NAME.zip",
        },
        "Handler": "__entrypoint__.handler",
        "MemorySize": 128,
        "Role": {
          "Fn::GetAtt": [
            "CustomSecurityHubBatchEnableStandardsCustomResourceProviderRole1ABC8ED2",
            "Arn",
          ],
        },
        "Runtime": "nodejs18.x",
        "Timeout": 900,
      },
      "Type": "AWS::Lambda::Function",
    },
    "CustomSecurityHubBatchEnableStandardsCustomResourceProviderLogGroup51D3B0EA": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "KmsKeyId": {
          "Fn::GetAtt": [
            "CustomKey1E6D0D07",
            "Arn",
          ],
        },
        "LogGroupName": {
          "Fn::Join": [
            "",
            [
              "/aws/lambda/",
              {
                "Ref": "CustomSecurityHubBatchEnableStandardsCustomResourceProviderHandler4BE622C1",
              },
            ],
          ],
        },
        "RetentionInDays": 3653,
      },
      "Type": "AWS::Logs::LogGroup",
      "UpdateReplacePolicy": "Delete",
    },
    "CustomSecurityHubBatchEnableStandardsCustomResourceProviderRole1ABC8ED2": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Sub": "arn:\${AWS::Partition}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
          },
        ],
        "Policies": [
          {
            "PolicyDocument": {
              "Statement": [
                {
                  "Action": [
                    "securityhub:BatchDisableStandards",
                    "securityhub:BatchEnableStandards",
                    "securityhub:DescribeStandards",
                    "securityhub:DescribeStandardsControls",
                    "securityhub:EnableSecurityHub",
                    "securityhub:GetEnabledStandards",
                    "securityhub:UpdateStandardsControl",
                  ],
                  "Effect": "Allow",
                  "Resource": "*",
                  "Sid": "SecurityHubCreateMembersTaskSecurityHubActions",
                },
                {
                  "Action": [
                    "iam:CreateServiceLinkedRole",
                  ],
                  "Condition": {
                    "StringLike": {
                      "iam:AWSServiceName": "securityhub.amazonaws.com",
                    },
                  },
                  "Effect": "Allow",
                  "Resource": "*",
                  "Sid": "SecurityHubServiceLinkedRole",
                },
              ],
              "Version": "2012-10-17",
            },
            "PolicyName": "Inline",
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "SecurityHubStandards294083BB": {
      "DeletionPolicy": "Delete",
      "DependsOn": [
        "CustomSecurityHubBatchEnableStandardsCustomResourceProviderLogGroup51D3B0EA",
      ],
      "Properties": {
        "ServiceToken": {
          "Fn::GetAtt": [
            "CustomSecurityHubBatchEnableStandardsCustomResourceProviderHandler4BE622C1",
            "Arn",
          ],
        },
        "region": {
          "Ref": "AWS::Region",
        },
        "standards": [
          {
            "controlsToDisable": [
              "IAM.1",
              "EC2.10",
              "Lambda.4",
            ],
            "enable": true,
            "name": "AWS Foundational Security Best Practices v1.0.0",
          },
          {
            "controlsToDisable": [
              "IAM.1",
              "EC2.10",
              "Lambda.4",
            ],
            "enable": true,
            "name": "PCI DSS v3.2.1",
          },
        ],
      },
      "Type": "Custom::SecurityHubBatchEnableStandards",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;
